sale_blanket_order/models/blanket_orders.py

```diff
@@ -307,7 +317,22 @@ def _compute_quantities(self):
             line.ordered_qty = sum(l.product_uom_qty for l in sale_lines)
             line.invoiced_qty = sum(l.qty_invoiced for l in sale_lines)
             line.delivered_qty = sum(l.qty_delivered for l in sale_lines)
-            line.remaining_qty = line.original_qty - line.ordered_qty
+            line.planned_qty = sum(l.next_order_qty for l in line.planned_line_ids)
+            line.remaining_qty = line.original_qty - line.ordered_qty - line.planned_qty
+
+    @api.multi
+    @api.depends(
+        'planned_line_ids.next_order_date'
+    )
+    def _compute_next_planned_order(self):
+        for line in self:
+            planned_lines = line.planned_line_ids
+            if planned_lines:
+                next_planned_line = planned_lines.sorted(key=lambda r: r.next_order_date)[0]
+                for lines in next_planned_line:
```

Comment:
What is the use of the for loop here? I think it needs to be removed.

Reply:
Sorry was used for debugging, my bad!
It has been removed